modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteScrollableHitSource.java

```diff
@@ -221,15 +222,21 @@ protected void doRun() throws Exception {
                 }
 
                 @Override
-                public void onFailure(Exception t) {
-                    if (t instanceof ResponseException) {
-                        ResponseException re = (ResponseException) t;
+                public void onFailure(Exception e) {
+                    if (e instanceof ResponseException) {
+                        ResponseException re = (ResponseException) e;
                         if (RestStatus.TOO_MANY_REQUESTS.getStatus() == re.getResponse().getStatusLine().getStatusCode()) {
-                            listener.onRetryableFailure(t);
+                            listener.onRetryableFailure(e);
                             return;
                         }
+                        if (RestStatus.UNAUTHORIZED.getStatus() == re.getResponse().getStatusLine().getStatusCode()) {
```

Comment:
I don't think `4xx -> 400`  and `5xx -> 500` is good enough. Sure we are against building new exceptions but then we are ok with losing info about errors here? I personally think we should make this work properly, not sure what others think.